app/src/main/java/com/example/gfood/screen/mealdetail/MealDetailActivity.kt

```diff
@@ -1,27 +1,103 @@
 package com.example.gfood.screen.mealdetail
 
+import android.annotation.SuppressLint
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
-import androidx.appcompat.app.AppCompatActivity
 import com.example.gfood.R
 import com.example.gfood.data.model.Meal
+import com.example.gfood.data.model.MealDetail
+import com.example.gfood.data.source.local.MealFavouriteLocalDataSource
+import com.example.gfood.data.source.remote.MealDetailRemoteDataSource
+import com.example.gfood.data.source.repository.MealDetailRepository
+import com.example.gfood.data.source.repository.MealFavouriteRepository
+import com.example.gfood.utils.Constant
+import com.google.android.youtube.player.YouTubeBaseActivity
+import com.google.android.youtube.player.YouTubeInitializationResult
+import com.google.android.youtube.player.YouTubePlayer
+import kotlinx.android.synthetic.main.activity_meal_detail.*
+
+class MealDetailActivity : YouTubeBaseActivity(), ContractMealDetail.View {
+
+    private var presenter: MealDetailPresenter? = null
+    private var meal: Meal? = null
+    private var mealDetail: MealDetail? = null
 
-class MealDetailActivity : AppCompatActivity() {
-    
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_meal_detail)
+        meal = intent.getParcelableExtra(BUNDLE_MEAL) as Meal?
+        initData()
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onGetMealDetailSuccess(listMealDetails: MutableList<MealDetail>) {
+        this.mealDetail = listMealDetails[0]
```

Comment:
trường hợp listMeal = null thì app sẽ crash thì sao e?
```suggestion
       mealDetail = listMealDetails[0]
```